File: solutions/c/collatz-conjecture/1/collatz_conjecture.c
#include "collatz_conjecture.h"

int steps(int start){
    if (start < 1) return ERROR_VALUE;
    int steps = 0;
    while (start != 1){
        if (start%2) start = 3*start + 1;
        else start /= 2;
        steps++;
    }
    return steps;
}

File: solutions/julia/collatz-conjecture/1/collatz-conjecture.jl
function collatz_steps(n)
    if n < 1
        throw(DomainError(n))
    end
    steps = 0
    while n != 1
        steps += 1
        if n % 2 == 0
            n /= 2
        else
            n = 3n + 1
        end
    end
    steps
end